### ISSUE TITLE
Upgrade to React 18

### ISSUE THREAD

**Issue by arcdev1** [arcnovus/wet-boew-x#21]
https://reactjs.org/blog/2022/03/08/react-18-upgrade-guide.html

**Comment by teaganryley** [comment]
Hi there, are there any plans to upgrade to React 18 in the near future? Thanks.

**Comment by arcdev1** [comment]
Yes ... in the next 2 months. What particular features of React 18 are you looking to adopt?

**Comment by teaganryley** [comment]
In particular, I'm interested in the automatic batching, the new renderer, the new strict mode behaviours, and transitions. Long-term I would adopt practically all of the new features, save for server rendering.

**Comment by teaganryley** [comment]
Hypothetically speaking, what would be the consequences of running this package with React/react-dom 18? I assume it's not advisable.

**Comment by arcdev1** [comment]
It's not well tested at the moment.
